src/llama.cpp

```diff
@@ -21139,6 +21139,10 @@ struct llama_batch_allocr {
     // optionally fulfill the batch returned by llama_batch_get_one
     llama_batch_allocr(struct llama_context * ctx, struct llama_batch in_batch) {
         batch = in_batch;
+        if (batch.n_tokens == 0) {
+            // llama_(de|en)code_internal will return an error in this case
+            return;
+        }
```

Comment:
My experience is that constructors should be as simple as possible and avoid branches if possible since they can leave the object in partially initialized state without a way to handle this. Suggest to replace this block with `assert()` and move `llama_batch_allocr` from `llama_encode`/`llama_decode` to `llama_..._internal` after the empty batch check.